$MirOS: ports/shells/zsh/patches/patch-Src_utils_c,v 1.1 2009/09/06 16:31:13 tg Exp $
--- Src/utils.c.orig	Mon Dec 20 10:00:30 2010
+++ Src/utils.c	Tue Mar  8 17:18:23 2011
@@ -537,11 +537,14 @@ wcs_nicechar(wchar_t c, size_t *widthp, 
 	 * This comparison fails and generates a compiler warning
 	 * if wchar_t is 16 bits, but the code is still correct.
 	 */
+#if !defined(__WCHAR_MAX__) || (__WCHAR_MAX__ > 65535U)
 	if (c >=  0x10000) {
 	    sprintf(buf, "\\U%.8x", (unsigned int)c);
 	    if (widthp)
 		*widthp = 10;
-	} else if (c >= 0x100) {
+	} else
+#endif
+	  if (c >= 0x100) {
 	    sprintf(buf, "\\u%.4x", (unsigned int)c);
 	    if (widthp)
 		*widthp = 6;
